federation/cluster/federation-up.sh

```diff
@@ -113,6 +115,31 @@ function join_clusters() {
   done
 }
 
+# debug_apiserver_service is a function purely used for debugging federation
+# apiserver service that occasionally times out due to failure in obtaining
+# loadbalancer address from GCP. This function should be deleted after
+# its purpose is served.
+function debug_apiserver_service() {
+  while ! svc_uid=$("${KUBE_ROOT}/cluster/kubectl.sh" \
+    --context="${HOST_CLUSTER_CONTEXT}" \
+    --namespace=${FEDERATION_NAMESPACE} \
+    -o jsonpath='{.metadata.uid}' \
+    get svc \
+    "${FEDERATION_NAME}-apiserver" 2>/dev/null); do
+    sleep 10
+  done
+
+  fwid_full="a${svc_uid//-/}"
```

Comment:
Yeah. Based on this - https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L62